temporallib/worker/sentry_interceptor.py

```diff
@@ -19,12 +20,21 @@
 
 @dataclass
 class SentryOptions:
-    dsn: str
+    """
+    Defines the parameters for configuring Sentry error reporting.
+    """
+    dsn: str = None
     release: str = None
     environment: str = None
-    sample_rate: float = 1.0
-    redact_params: bool = False
-
+    sample_rate: float = None
+    redact_params: bool = None
+
+    def __post_init__(self):
```

Comment:
DataClasses are well suited for simplicity and being lightweight(especially with huge data) but for this case, I think we should use `Pydantic`. Rather than adding custom logic in `__post_init` you can simply benefic from `Pydantic`.

Reply:
Great suggestion. I've refactored these classes to use Pydantic instead.